From dbbc3d3a5b44ce7a75a91eaf2613fe88c92a29d8 Mon Sep 17 00:00:00 2001
From: Sergio Schvezov <sergio.schvezov@canonical.com>
Date: Wed, 11 Dec 2013 23:30:48 -0300
Subject: [PATCH] The loop search for .go files returns incorrectly

In the copy part of configure when looping through the sourcefiles and
not exporting DH_GOLANG_INSTALL_ALL all the go files are incorrectly
skipped.

Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>
---
 lib/Debian/Debhelper/Buildsystem/golang.pm | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/Debian/Debhelper/Buildsystem/golang.pm b/lib/Debian/Debhelper/Buildsystem/golang.pm
index 9d55e63..c73c387 100644
--- a/lib/Debian/Debhelper/Buildsystem/golang.pm
+++ b/lib/Debian/Debhelper/Buildsystem/golang.pm
@@ -65,7 +65,7 @@ sub configure {
         },
         wanted => sub {
             my $name = $File::Find::name;
-            return if substr($name, -3) eq '.go' &&
+            return if !substr($name, -3) eq '.go' &&
                       (!exists($ENV{DH_GOLANG_INSTALL_ALL}) ||
                        $ENV{DH_GOLANG_INSTALL_ALL} != 1);
             return unless -f $name;
-- 
1.8.5.1

